### ISSUE TITLE
`std::auto_ptr` is deprecated since C++11

### ISSUE THREAD

**Issue by e-kwsm** [IUPAC-InChI/RInChI#25]
https://github.com/IUPAC-InChI/RInChI/blob/d122a78457c592b9728906f3c0b565a2c2c5d6dd/src/parsers/rinchi_reader.cpp#L101

https://github.com/IUPAC-InChI/RInChI/blob/d122a78457c592b9728906f3c0b565a2c2c5d6dd/src/parsers/rinchi_reader.cpp#L392

https://github.com/IUPAC-InChI/RInChI/blob/d122a78457c592b9728906f3c0b565a2c2c5d6dd/src/rinchi/rinchi_reaction.cpp#L160

**Comment by janholstjensen** [comment]
Correct. We should get those changed to std::unique_ptr(). Thanks for escalating this to an issue so we can get it done :-).
